### DISCUSSION TITLE
Multiple FlashRuntimeExtensions.so in the apk

### DISCUSSION THREAD

**Discussion by manishpatel0404** [airsdk/Adobe-Runtime-Support#1007] [Q&A] [answered]
Hi,

I am facing a peculiar problem while packaging an APK. As one can see in the attached image, the APK contains two copies of the file FlashRuntimeExtensions.so in the lib\arm64-v8a folder of the APK. One for each of the two test ANEs I have built. The following is the adt call, I used:

`adt -package -target apk -arch armv7 %SIGNING_OPTIONS% "%OUTPUT%" %APP_XML% %FILE_OR_DIR% -extdir "%EXT_DIR%"`

![APK_Image](https://user-images.githubusercontent.com/20116584/126919716-c9dcd4f7-ff27-48c4-9226-1e03c5bdf952.png)

FlashDevelop gave me an "Invalid APK" error. When I tried to install the APK on an Android phone, I got an “Problem parsing the package" error. When I investigated the APK, I noticed the two copies of the FlashRuntimeExtensions.so file.

When I manually delete one of these files from the APK, it installs as expected and the ANEs also work as expected.

The same problem occurs with debug and captive-runtime targets, as well and with any/all combinations of armv7, armv8 and x86 architectures.

If it is of any consequence, I am using AIR SDK version 33.1.476

Surprisingly, I found no help from an online search, which leads me to believe that I must be doing something stupid. Or, is it a bug?

Can someone throw some light on this? Thanks.

**Comment by ajwfrost** [comment, marked as answer]
FlashRuntimeExtensions.so shouldn't be packaged up with an ANE, it's only there to give you something to link against..  
I had actually thought that the packaging process would eliminate multiple libraries with the same name but maybe that was only added for .jar files...

But the .so file there shouldn't be included in the final package, it won't be needed because the symbols (FRE function implementations) come from the AIR library that's loaded first.

thanks

**Reply by manishpatel0404** [reply]
Thanks, once again, @ajwfrost 

I packaged the ANEs without the FlashRuntimeExtensions.so files, re-created the APK and everything works well on all android targets, as well as all architectures.

I was following the official Adobe documentation air-extensions.pdf which led me to create these extra folders and copy the FlashRuntimeExtensions.so in each of these folders. Wish the documentation was updated.

Anyway, everything works now and I am happy.

Thanks, once again.
